tests/compile-fail/cast_fn_ptr_unsafe.rs

```diff
@@ -0,0 +1,10 @@
+// just making sure that fn -> unsafe fn casts are handled by rustc so miri doesn't have to
```

Comment:
I realized just after I posted that, this would matter for implementing casting. I guess that's fine.